src/client/Microsoft.Identity.Client/ManagedIdentity/V2/ImdsV2ManagedIdentitySource.cs

```diff
@@ -251,12 +254,150 @@ private async Task<CertificateRequestResponse> ExecuteCertificateRequestAsync(st
+                .Replace("\r", "")
+                .Replace("\n", "")
+                .Replace(" ", "");
 
-            throw new NotImplementedException();
+            if (string.IsNullOrEmpty(base64Content))
+            {
+                throw new ArgumentException("Invalid PEM format: no certificate content found", nameof(certificatePem));
+            }
+
+            try
+            {
+                byte[] certBytes = Convert.FromBase64String(base64Content);
+                return new X509Certificate2(certBytes);
+            }
+            catch (FormatException ex)
+            {
+                throw new FormatException("Invalid PEM format: certificate content is not valid Base64", ex);
+            }
+        }
+
+        /// <summary>
+        /// Attaches a private key to a certificate for older .NET Framework versions.
+        /// This method uses the older RSACng approach for .NET Framework 4.7.2 and .NET Standard 2.0.
+        /// </summary>
+        /// <param name="certificate">The certificate without private key</param>
+        /// <param name="privateKey">The RSA private key to attach</param>
+        /// <returns>An X509Certificate2 with the private key attached</returns>
+        /// <exception cref="NotSupportedException">Thrown when private key attachment fails</exception>
+        internal X509Certificate2 AttachPrivateKeyToOlderFrameworks(X509Certificate2 certificate, RSA privateKey)
+        {
+            try
+            {
+                // For older frameworks, we need to use the legacy approach with RSACryptoServiceProvider
+                // First, export the RSA parameters from the provided private key
+                var parameters = privateKey.ExportParameters(includePrivateParameters: true);
+                
+                // Create a new RSACryptoServiceProvider with the correct key size
+                int keySize = parameters.Modulus.Length * 8;
+                var rsaProvider = new RSACryptoServiceProvider(keySize);
+                
+                try
+                {
+                    // Import the parameters into the new provider
+                    rsaProvider.ImportParameters(parameters);
+                    
+                    // Create a new certificate instance from the raw data
+                    var certWithPrivateKey = new X509Certificate2(certificate.RawData);
+                    
+                    // Assign the private key using the legacy property
+                    certWithPrivateKey.PrivateKey = rsaProvider;
+                    
+                    return certWithPrivateKey;
+                }
+                catch
+                {
+                    // Clean up the RSA provider if something goes wrong
+                    rsaProvider?.Dispose();
+                    throw;
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new NotSupportedException(
```

Comment:
I am not sure about this, I would just let the original exception propagate. Net462 should never reach this code, they should get an nicer exception earlier in the code.